src/java.net.http/share/classes/jdk/internal/net/http/HttpClientImpl.java

```diff
@@ -1686,6 +1690,18 @@ BufferSupplier getSSLBufferSupplier() {
         return sslBufferSupplier;
     }
 
+    private Duration getIdleConnectionTimeoutProp() {
+        String s = Utils.getNetProperty("jdk.httpclient.keepalive.timeout");
+        if (s != null) {
+            try {
+                long val = Long.parseLong(s);
+                if (val >= 0)
+                    return Duration.ofSeconds(Long.parseLong(s));
+            } catch (NumberFormatException ignored) {}
+        }
+        return null;
+    }
```

Comment:
If the property is not present or doesn't parse to an Integer value then the default value should be used. `Utils.getIntegerNetProperty` - which ConnectionPool uses, does that. Possibly we should also define what happens is the value is 0 or negative: we should probably be using the default value in that case too.

The property should also be read only once - like in ConnectionPool. I believe the KEEP_ALIVE constant that is currently in ConnectionPool should be moved here. Then its value can then be shared by this code & ConnectionPool. Not sure whether we'd need an `Optional<Duration> idleConnectionTimeout() ` method any more then. The IdleConnectionTimeout could just be initialized with `Duration.ofSeconds(HttpClientImpl.KEEP_ALIVE)`. Or possibly we could have a `Duration idleConnectionTimeout()` method.

It would be good to refactor ConnectionPool and Http2Connection to use the same code for getting the HttpTimeout value - and if we expose a method that returns a Duration then maybe ConnectionPool should use that too.